### PR TITLE
Remove API deprecated in 3.1

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

At least, start with it. There's much more to do. It's a bit tedious because we need appropriate API removal notes.

### CODE REVIEW

File: doc/api/next_api_changes/removals.rst [hunk 1]
```diff
@@ -1,2 +1,70 @@
...
+  functions to plot the loaded data.)
+- ``rcsetup.validate_qt4()`` (no replacement)
+- ``rcsetup.validate_qt5()`` (no replacement)
+- ``rcsetup.validate_verbose()`` (no replacement)
+- ``rcsetup.ValidateInterval`` (no replacement)
+
+- ``sphinxext.plot_directive.plot_directive()``
+  (use the class ``PlotDirective`` instead)
+
+- ``Axes.aname`` property (no replacement)
```
Comment (anntzer) [inline doc/api/next_api_changes/removals.rst]: Artist.aname

File: lib/matplotlib/backends/backend_ps.py [hunk 2]
```diff
@@ -637,10 +580,6 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
 grestore
 """)
 
-    def new_gc(self):
```
Comment (anntzer) [inline lib/matplotlib/backends/backend_ps.py]: no?

File: doc/api/next_api_changes/removals.rst [hunk 3]
```diff
@@ -1,2 +1,70 @@
...
+  (use the class ``PlotDirective`` instead)
+
+- ``Artist.aname`` property (no replacement)
+- ``Axis.iter_ticks`` (no replacement)
+
+- ``image.BboxImage.interp_at_native`` property (no replacement)
+- ``lines.Line2D.verticalOffset`` property (no replacement)
+- ``bezier.find_r_to_boundary_of_closedpath()`` (no relacement)
+
+- ``quiver.Quiver.color()`` (use ``Quiver._getfacecolor()`` instead)
```
Comment (dstansby) [inline doc/api/next_api_changes/removals.rst]: I'm fairly sure we shouldn't be recommending anything private as replacements 😉 
```suggestion
- ``quiver.Quiver.color()`` (use ``Quiver.get_facecolor()`` instead)
```

Comment (timhoffm) [reply doc/api/next_api_changes/removals.rst]: Sure.

**Pull request merged**